bin/ScanPrep.py

```diff
@@ -0,0 +1,253 @@
+        scan_step: size of step between scan patches
+        batch_size: number of patches per batch
+        scan_group_size: number of batches to be handeled by each scan process
+        scan_ds: number of times to downsample befor scanning (scan patch will be 1/2^(scan_ds) the resolution of the initial image)
+        roi_mask_ds: number of times downssampled before identifying the regions to be scanned
+        complete_scan: flag indicating if the whole image should be scanned ignoring ROIs
+        compute_rois: flag indicating if existing ROIs should be recomputed
+        roi_identifier_model: what model to use for identifying ROIs if not precomputed and compute_rois
+        roi_thresh: Threshold for simple ROI detection models
+        scan_mask : create a mask of which patches have been scanned (useful for sanity checking the scan but creates an ineficent zarr)
+    """
+    image_path=f"{image_dir}slide_{image_id}.zarr"
+    roi_mask_path=f"{roi_mask_dir}roi_masks_{image_id}.zarr"
+    feature_path=f"{feature_dir}ViT_features_{image_id}.zarr"
+
+    #pre-create zarr array to contain feature scan (prevents race condition issiues that would arise if initialized at scan time)
+    zi=zarr.open(image_path)
+    zi=zi["0"]
+    zf=zarr.open(feature_path, mode='a')
+    zf.zeros(f"Features/{scan_ds}",shape=(vit_model_channels,int(np.ceil(zi[f"{scan_ds}"].shape[3]/scan_step)),int(np.ceil(zi[f"{scan_ds}"].shape[4]/scan_step))),chunks=(vit_model_channels,1,1),overwrite=True)
+    zf.zeros(f"ScanMask/{scan_ds}",shape=(int(np.ceil(zi[f"{scan_ds}"].shape[3]/scan_step)),int(np.ceil(zi[f"{scan_ds}"].shape[4]/scan_step))),chunks=(1,1),overwrite=True)
+
+    # compute roi mask if nessasary or indicated by parameters
+    zm=zarr.open(roi_mask_path)
+    if compute_rois or (f"{roi_type}/{roi_mask_ds}" not in zm):
+        if roi_identifier_model=="ChannelMeanThresh":
+            id_model=GuasianTissueThresholder(roi_thresh,[0,1,2],lambda x:np.mean(x,axis=0),10)
+        elif roi_identifier_model=="GreenThresh":
+            id_model=GuasianTissueThresholder(roi_thresh,[1],lambda x:x.squeeze(),10)
+        else:
+            raise ValueError("roi_identifier_model must be one of ['ChannelMeanThresh','GreenThresh']")
+        image_ds=zarr.open(image_path)[f'0/{roi_mask_ds}']
+        roi_mask=id_model(image_ds)
+        zm.array(f"{roi_type}/{roi_mask_ds}",roi_mask,overwrite=True)   
+    else:
+        roi_mask=zm[f"{roi_type}/{roi_mask_ds}"][:,:]
+    
+    # find roi regions
+    if complete_scan:
+        sections=pd.DataFrame.from_dict({"bbox-0":[0],"bbox-1":[0],"bbox-2":[zi[f"{scan_ds}"].shape[3]],"bbox-3":[zi[f"{scan_ds}"].shape[4]]})
+    else:
+        sections=findRegions(roi_mask,roi_min_area,roi_max_area)
+
+    #create scan group csvs
+    scanBatching(image_id, sections, scan_step, batch_size, scan_group_size, scan_ds, roi_mask_ds)
+
+
+if __name__=="__main__":
+    CLI(prepForViT)
+
+
```

Comment:
I have a test folder set up that is currently gitignored because I wanted to have a discussion about testing best practices and test data storage before putting any of that in the official repo.